packages/clerk-js/src/utils/meetsRequirement.ts

```diff
@@ -0,0 +1,34 @@
+import { Clerk, ClerkOptions, EnvironmentResource } from '@clerk/types';
+
+export type Requirements = 'singleSession' | 'user' | 'organization';
+export type MeetsRequirement = (
+  clerk: Clerk,
+  environment?: EnvironmentResource | null,
+  options?: ClerkOptions,
+) => boolean;
+
+export const meetsRequirement: Record<Requirements, MeetsRequirement> = {
```

Comment:
I definitely like the idea! Couple of comments:

1. You want to add a namespace over a group of exports, rather than using an explicit type and the dictionary, you can just: 

```
const hello = () => {}
const there = () => {}
export const ClicheStarwarsMeme = { hello, there }

// somewhere else
import { ClicheStarwarsMeme } from './'
ClicheStarwarsMeme.hello()
```

2. The `meetsRequirement` namespace makes sense when reading in in the context of the calling function, but does it make sense as a module on its own? What does `import { meetsRequirements }` mean? I think its a little generic. That's why **I think** I preferred your previous approach with the standalone, composable helpers 

LMK what you think!